zerver/management/commands/deliver_scheduled_messages.py

```diff
@@ -30,6 +31,52 @@ class Command(BaseCommand):
 Usage: ./manage.py deliver_scheduled_messages
 """
 
+    def should_deliver(self, message: ScheduledMessage) -> bool:
```

Comment:
This function should probably live in a new `zerver/lib/reminders.py` and have tests for the various cases.  

Generally, we try to avoid having significant code inside management commands, since it's hard to do automated tests for them.